server/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -232,30 +231,30 @@ static void addCollectorsAndSearch(SearchContext searchContext) throws QueryPhas
                 final Collector collector = collectorManager.newCollector();
                 collectorManager = wrapWithProfilerCollectorManagerIfNeeded(
                     searchContext.getProfilers(),
-                    new FilteredCollector(collector, filterWeight),
+                    new SingleThreadCollectorManager(new FilteredCollector(collector, filterWeight)),
                     REASON_SEARCH_POST_FILTER,
-                    collector
+                    (InternalProfileCollector) collector
                 );
             }
             if (searchContext.getAggsCollectorManager() != null) {
                 final Collector collector = collectorManager.newCollector();
                 final Collector aggsCollector = searchContext.getAggsCollectorManager().newCollector();
                 collectorManager = wrapWithProfilerCollectorManagerIfNeeded(
                     searchContext.getProfilers(),
-                    MultiCollector.wrap(collector, aggsCollector),
+                    new SingleThreadCollectorManager(MultiCollector.wrap(collector, aggsCollector)),
                     REASON_SEARCH_MULTI,
-                    collector,
-                    aggsCollector
+                    (InternalProfileCollector) collector,
+                    (InternalProfileCollector) aggsCollector
```

Comment:
I think this is true only when profiling?

Reply:
yes my bad, I need to restore what I had before. I went back to using collectors here, as we don't need managers yet. sorry about the back and forth,